SciLean/Data/Mesh/SurfaceMesh.lean

```diff
@@ -0,0 +1,350 @@
+/-
+Copyright (c) 2023 Siddharth Bhat. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Siddharth Bhat
+
+Surface Mesh based on discrete differential geometry.
+https://github.com/GeometryCollective/geometry-processing-js/blob/c7ea47ae808979a5361e3fcd61bf921a194bf897/core/mesh.js#L3-L213
+-/
+
+import SciLean.Core
+
+
+-- abbrev V3 : Type := ℝ^{3}
+-- 3D vector.
+structure V3 where
```

Comment:
In this file I have an attempt for short array optimization for ℝ^{n}, n=0,1,2,3,4. It was working at some point but recently it broke and was giving me too many headaches.

https://github.com/lecopivo/SciLean/blob/master/SciLean/Data/DataArray/ShortOptDataArray.lean

Any opinion on that, do you think it is a good idea for ℝ^{n} be a small struck for small n and array otherwise?

Also using ℝ^{n} for small n is pain in the ass, you can't easily create a vector (x,y,z).